webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,234 @@
+          values.name,
+          values.pool,
+        );
+      }
+      onLaunch();
+    },
+    [ showFullConfig, form, onLaunch ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((e) => {
+    if (e === '') {
+      setResourceTypeOptions([ { id:'CPU', label:'CPU' }, { id:'GPU', label:'GPU' } ]);
+    } else {
+      const pool = resourcePools.find(pool => pool.name === e);
+      if (pool){
+        const options = [];
+        if (pool.cpuContainerCapacityPerAgent > 0) {
+          options.push({ id:'CPU', label:'CPU' });
+        }
+        if (pool.slotsPerAgent && pool.slotsPerAgent > 0) {
+          options.push({ id:'GPU', label:'GPU' });
+        }
+        setResourceTypeOptions(options);
+        form.setFieldsValue({ type:undefined });
+        setResourceType(undefined);
+      }
+    }
+  },[ resourcePools, form ]);
+
+  const handleTypeUpdate = useCallback((e) => {
+    setResourceType(e);
+  },[]);
+
+  return <Modal
+    footer={<>
+      <Button onClick={handleSecondary}>{showFullConfig ? 'Edit Form' : 'Edit Full Config'}</Button>
+      <Button
+        type="primary"
+        onClick={() => {
+          form.validateFields().then(values => {
+            handleCreateEnvironment(values);
+          }).catch();
+        }
+        }>Launch</Button>
+    </>}
+    title='Launch JupyterLab'
+    visible={visible}
+    {...props}>
+    {showFullConfig?
+      <Form form={form}>
+        <div style={{
+          backgroundColor:'rgb(230,230,230)',
+          border:'1px solid rgb(200,200,200)',
+          marginBottom: '4px',
+          padding: 2,
+          textAlign:'center',
+        }}
+        >
+          <Link external path="/docs/reference/command-notebook-config.html">
+          Read about notebook settings
+          </Link>
+        </div>
+        <Item
+          name='config'
+          rules={[ { message: 'Invalid YAML', required: true }, () => ({
+            validator(_, value) {
+              try {
+                yaml.load(value);
+                return Promise.resolve();
+              } catch(err) {
+                return Promise.reject(new Error('Invalid YAML'));
+              }
+            },
+          }) ]}>
+          <MonacoEditor
+            height={400}
+            language='yaml'
+            options={{
+              minimap: { enabled: false },
+              scrollBeyondLastLine: false,
+              selectOnLineNumbers: true,
+            }} />
+        </Item>
+      </Form> :
+      <Form
+        form={form}
+        initialValues={storage.getWithDefault(STORAGE_KEY, {
+          pool: '',
+          slots:1,
+          template: '',
+          type: undefined,
+        })}
+        labelCol={{ span:8 }}
+        onValuesChange={storeConfig}>
+        <Item label='Notebook Template' name='template'>
+          <Select
+            style={{ minWidth:120 }}>
+            <Option key='' value=''>---Empty---</Option>
```

Comment:
**suggestion:** instead of an empty option, use the `placeholder` prop on `Select`. something like `placeholder="Select a notebook template"`

Reply:
Choosing the empty option has semantic meaning in this case - it means that you want to use the default template. Using a placeholder would mean that if you choose a template and then change your mind, you wouldn't be able to go back to the default, as the placeholder disappears once something else is chosen.

Reply:
I see. If that's the case, I would recommend the option label as `Default Task Template` or something to that effect so the user gets a bit more info of what they are selecting.